### PR TITLE
update hello world tutorial

### PR DESCRIPTION
**Pull request by jinglescode**
@KtorZ, I made a few changes here:
- update hello world tutorial to use `MeshTxBuilder` instead of `Transaction` for future proofing
- after Chang, reference script fee calculation requires fetcher
- use TypeScript instead
- added a link to the code example so anyone and test
- remove tangocrypto, its not live anymore

### CODE REVIEW

File: src/pages/example--hello-world/end-to-end/mesh.mdx [hunk 1]
```diff
@@ -85,171 +77,214 @@ Now that we have some funds, we can lock them in our newly created contract. We'
...
+  serializePlutusScript,
+  UTxO,
+} from "@meshsdk/core";
+import { applyParamsToScript } from "@meshsdk/core-csl";
 
-const blockchainProvider = new BlockfrostProvider(process.env.BLOCKFROST_PROJECT_ID);
+// BlockfrostProvider to query data from the blockchain
+const blockchainProvider = new BlockfrostProvider(
+  fs.readFileSync("bf.key").toString()
+);
```
Comment (KtorZ) [inline src/pages/example--hello-world/end-to-end/mesh.mdx:96]:
```suggestion
const blockchainProvider = new BlockfrostProvider(process.env.BLOCKFROST_PROJECT_ID);
```

Let's keep the ENV var here, thanks 🙏

File: src/pages/example--hello-world/end-to-end/mesh.mdx [hunk 2]
```diff
@@ -85,171 +77,214 @@ Now that we have some funds, we can lock them in our newly created contract. We'
...
 
-<Callout type="warning">
-Note that the highlighted line above looks for an environment variable named `BLOCKFROST_PROJECT_ID` which its value must be set to your Blockfrost project id.
-You can define a new environment variable in your terminal by running (in the same session you're also executing the script!):
+Note that the highlighted line above looks Blockfrost project ID, so we create a file named `bf.key` and add your Blockfrost project ID to it.
 
-```console
-export BLOCKFROST_PROJECT_ID=preview...
+```console filename="bf.key"
+preview...
```
Comment (KtorZ) [inline src/pages/example--hello-world/end-to-end/mesh.mdx:113]:
```suggestion
export BLOCKFROST_PROJECT_ID=preview...
```

File: src/pages/example--hello-world/end-to-end/mesh.mdx [hunk 3]
```diff
@@ -85,171 +77,214 @@ Now that we have some funds, we can lock them in our newly created contract. We'
...
+    type: "root",
+    bech32: fs.readFileSync("me.sk").toString(),
   },
 });
 ```
 
-<Callout type="warning">
-Note that the highlighted line above looks for an environment variable named `BLOCKFROST_PROJECT_ID` which its value must be set to your Blockfrost project id.
-You can define a new environment variable in your terminal by running (in the same session you're also executing the script!):
+Note that the highlighted line above looks Blockfrost project ID, so we create a file named `bf.key` and add your Blockfrost project ID to it.
```
Comment (KtorZ) [inline src/pages/example--hello-world/end-to-end/mesh.mdx:110]:
```suggestion
<Callout type="warning">
Note that the highlighted line above looks for an environment variable named `BLOCKFROST_PROJECT_ID` which its value must be set to your Blockfrost project id.
You can define a new environment variable in your terminal by running (in the same session you're also executing the script!):
```

File: src/pages/example--hello-world/end-to-end/mesh.mdx [hunk 4]
```diff
@@ -85,171 +77,214 @@ Now that we have some funds, we can lock them in our newly created contract. We'
...
 
-```console
-export BLOCKFROST_PROJECT_ID=preview...
+```console filename="bf.key"
+preview...
 ```
 
 Replace `preview...` with your actual project id.
-</Callout>
```
Comment (KtorZ) [inline src/pages/example--hello-world/end-to-end/mesh.mdx:117]:
```suggestion
</Callout>

```

File: src/pages/example--hello-world/end-to-end/mesh.mdx [hunk 5]
```diff
@@ -85,171 +77,214 @@ Now that we have some funds, we can lock them in our newly created contract. We'
...
+  serializePlutusScript,
+  UTxO,
+} from "@meshsdk/core";
+import { applyParamsToScript } from "@meshsdk/core-csl";
 
-const blockchainProvider = new BlockfrostProvider(process.env.BLOCKFROST_PROJECT_ID);
+// BlockfrostProvider to query data from the blockchain
+const blockchainProvider = new BlockfrostProvider(
+  fs.readFileSync("bf.key").toString()
+);
```
Comment (KtorZ) [inline src/pages/example--hello-world/end-to-end/mesh.mdx:134]:
```suggestion
const blockchainProvider = new BlockfrostProvider(process.env.BLOCKFROST_PROJECT_ID);
```

File: src/pages/example--hello-world/end-to-end/mesh.mdx [hunk 6]
```diff
@@ -85,171 +77,214 @@ Now that we have some funds, we can lock them in our newly created contract. We'
...
```
Comment (KtorZ) [inline src/pages/example--hello-world/end-to-end/mesh.mdx:180]:
```suggestion
const blockchainProvider = new BlockfrostProvider(process.env.BLOCKFROST_PROJECT_ID);
```

File: src/pages/example--hello-world/end-to-end/mesh.mdx [hunk 7]
```diff
@@ -325,105 +361,84 @@ to unlock.
...
-    Tx ID: 1e1987e0d0aaa35a631bdc0a0dfbadcba5b33b349d1153612cac4167262fdae2
-    Redeemer: {"data":{"alternative":0,"fields":["Hello, World!"]}}
+1 tADA unlocked from the contract at Tx ID: af3e7d83e5ee5324a612de9126636ef55505ffb9c468dacaf419635921b7a91c
 ```
 
 And, tada 🎉!
 
 We can inspect our [redeeming transaction on CardanoScan](https://preview.cardanoscan.io/transaction/d3d5e828a3989691b0960d22a265c8c9ae4723134b52aa05ec0fb7d40f060392?tab=contracts) and see that it successfully executed our _Hello World_ contract.
 
+You can find the [full source code](https://github.com/MeshJS/examples/tree/main/aiken-hello-world) for this tutorial on GitHub.
```
Comment (KtorZ) [inline src/pages/example--hello-world/end-to-end/mesh.mdx:440]:
```suggestion
You can find the [full source code](https://github.com/MeshJS/examples/tree/main/aiken-hello-world) for this tutorial on MeshJS's GitHub.
```

**Review by KtorZ — CHANGES_REQUESTED** [review]
Thanks for this @jinglescode, it was on my TODO list lately. It looks good, except for the `bf.key` file. Please do not encourage people to write their private API keys on the filesystem, especially in a git working tree :') ... 

So I reverted those changed back to using an ENV var; which likely messes up a bit with the code highlight indices. Haven't re-checked those.

**Comment by jinglescode** [comment]
looking good. thanks for reviewing.

**Review by KtorZ — APPROVED** [review]

**Comment by jinglescode** [comment]
@KtorZ the live site is not updated.

**Pull request merged**